## S1. Comparative Analysis of shinySRT and Other SRT Interactive Visualization Tools
### S1.1 SRT Visualization Tools Selected for Comparative Analysis.
Our tool is developed to visualize SRT data and facilitate interactive analysis and data sharing for researchers without a bioinformatics background. Below is a comparative list between shinySRT and existing SRT data visualization tools, we present a compilation of 9 SRT sequencing data visualization and analysis tools: SMDB(Cao et al., 2023), spatialLIBD (Pardo et al., 2022), Cellxgene VIP (Kejie et al., 2022), Samui (Chaichontat et al., 2023), SpatialView (Chitrasen et al., 2023), STutility (Bergenstråhle et al., 2020), Loupe Browser, SPATA2 (Jan et al., 2020), TissUUmaps3 (Pielawski et al., 2023), and i-stLearn (Duy et al., 2023). Some tools possess the capability to execute complete data workflow and SRT sequence data visualization, including STutility, TissUUmaps3, and i-stLearn. 


Table 1: Comparison of several SRT spatial resolved visualization tools in terms of key functionalities (yellow rows), display / UI options (blue rows), supported input file formats (green rows). Further detailed explanations will follow.

ShinySRT and other similar tools are compared in the following three aspects: (1) key functionalities, (2) supported input file formats, and (3) display methods/UI options (Table 1). 


 
### S1.2 Core Features of Diverse Packages
Firstly, compare the key functionalities:
-	Web sharing refers to the capability of hosting and disseminating visualized data on a webpage. Among these tools listed, both shinySRT and the other five support online sharing, facilitating researchers without a bioinformatics background to collaboratively share SRT data on the web. 
-	ST analysis refers to spatially resolved data analysis, including data standardization, dimensionality reduction, and clustering. STutility, TissUUmaps3, and i-stLearn support SRT analysis. SPATA2 does not support dimensionality reduction but facilitates trajectory and CNV analysis. The remaining tools, including shinySRT, are specialized for the visualization of SRT sequencing data. As there are already comprehensive tools available for SRT sequencing data analysis, we do not possess capability of analytical functionalities related to SRT analysis. 
-	The presence of Docker signifies whether the tool includes Docker images, streamlining the deployment and execution of applications across diverse machines and platforms. Due to ShinySRT being primarily built upon the Shiny framework, with its interface directly accessible on the web, and considering the convenient installation process of ShinySRT, therefore, this function is not available. 
-	Cloud refers to the capability of deploying web applications on public cloud platforms. ShinySRT supports this functionality, utilizing the R Shiny backend for seamless online deployment. 
-	Software as a service (SaaS) refers to the inclusive hosting capabilities of a tool, allowing users to upload data and directly utilize the tool as a service. 
-	Selecting area refers to the ability to perform direct interactive operations on spatial images, such as region selection, enabling the analysis and comparison of gene expression within specific local regions. While all these tools allow direct point selection on spatial images, only shinySRT, SMDB, Cellxgene VIP, STutility, TissUUmaps3, and i-stLearn have the capability to analyze gene expression within the selected regions. 
-	Single-cell resolution refers to the capability of supporting the visualization of spatial transcriptomics data with single-cell precision, including image-based spatial data. 

### S1.3 User Interface Options Across Various Packages
The following is a comparison of the data visualization modules in different tools.
-	Multiple comparison: Displaying spatial metadata and gene expression for multiple samples within the same interface. Most tools can only display gene expressions for a single sample or struggle to concurrently present the expression profiles of multiple samples. While some tools allow for the importation of multiple samples, existing spatial transcriptomics visualization tools lack the capability to simultaneously visualize the spatial expression of genes across multiple samples. Consequently, cross-sample comparisons become challenging. ShinySRT not only enables the importation of multiple spatial transcriptomics datasets but also facilitates their display on the same page, offering researchers a convenient platform for cross-sample analysis. 
-	Zoom In/Out: In an interactive interface, this feature allows users to zoom in or out on specific regions of the main plot. Leveraging ggiraph, shinySRT, like other tools, enables spatial zooming in and out for a more detailed exploration. 
-	Gene Coexpression: This feature displays the expression patterns of two genes on the same spatial map. This functionality is unique to shinySRT and Samui. In contrast, most tools typically allow the visualization of only one gene at a time in the spatial context. Additionally, shinySRT provides the flexibility to adjust gene expression thresholds, allowing users to fine-tune the display of gene expression patterns. 
-	Proportion Plots: These plots illustrate the composition of spots within different spot groups, indicating the proportion of spots from distinct regions. For example, in the clustering results obtained from Seurat, proportion plots can depict the distribution of spots from various areas within different clusters. 
-	Multi-Gene Bubble plot/Heatmap: This visualization method utilizes a heatmap or bubble plot to depict the expression patterns of multiple genes within different groups. 
-	Differential Expression: Variance analysis. Typically, visualization tools often lack built-in differential analysis functionality, as the process is computationally intensive and requires substantial memory resources. ShinySRT, designed as a lightweight tool, does not include this feature. Moreover, there are already numerous existing tools specifically designed for conducting differential expression analysis. 
-	Export Images as PDF/PNG: This feature allows users to export spatial maps and other statistical plots. ShinySRT supports the export of files in both PDF and PNG formats, facilitating their use for presentations, manuscripts, with the additional capability to adjust image sizes. 

### S1.4 Compatible Input File Formats Supported
The capability to support multiple data formats becomes crucial for visualization tools. Below, we will compare the various tools in terms of their support for different data types. For SRT sequencing data, common data formats include: 
-	Unprocessed files represent a ubiquitous plain-text format utilized for storing gene expression matrices and SRT metadata. 
-	The h5ad format is a common data type used in Python for spatial transcriptomics analysis. It enables the storage of large datasets without occupying memory during retrieval. This format is typically employed in Python-based spatial transcriptomics analysis tools, e.g. SCANPY. 
-	SCE (SingleCellExperiment) object is a data structure in the R programming language designed for storing and processing single-cell RNA sequencing data. This object is commonly utilized for integrating and analyzing expression data at the single-cell level, encompassing gene expression, sample information, and other pertinent metadata. The SCE object serves as a convenient tool for organizing and manipulating single-cell data, facilitating various analyses such as differential expression analysis, clustering, and visualization. Additionally, SCE can also be employed for storing Spatial Transcriptomics (SRT) data, and the SRT data format provided by spatialLIBD is in the SCE format. 
-	SpatialExperiment object is a data structure in the R programming language designed for storing spatial omics experimental data. It represents an extension of the SCE object, originally designed for single-cell data, with the specific purpose of supporting the storage and retrieval of additional information from both spot-based and molecular-based platforms. 
-	Seurat object is a data structure defined by the Seurat package in the R programming language, generated exclusively by Seurat. It is primarily employed for the storage and analysis of single-cell RNA sequencing (scRNA-seq) data as well as spatial transcriptomics (SRT) sequencing data. 
ShinySRT supports various common spatial transcriptomics data formats, including h5ad, SPE, SCE, Seurat, as well as expression matrices provided in unprocessed formats. Moreover, it is compatible with several other data formats. As an R package, users can convert various data formats into list structures in R by following the provided examples. and subsequently transform them into interactive spatial transcriptomics interfaces using shinySRT. 

Cellxgene VIP, Samui, and SpatialView offer powerful interactive and cloud-based visualization capabilities. Cellxgene VIP supports the visualization of single-cell RNA sequencing (scRNA-seq) and spatial transcriptomics (SRT) data, allowing users to view and analyze these datasets on platform. It is comprehensive and robust, excelling in the visualization of single-cell and SRT data, and offers a variety of primary data visualization methods. Additionally, it includes interactive statistical analysis features, such as differential analysis. Cellxgene VIP supports reading processed h5ad objects but cannot display SRT data on the same interface or perform cross-sample analysis. Due to its extensive features, the interface is relatively complex and requires a detailed user manual for onboarding. 

Samui is a sleek and visually appealing tool for SRT data visualization, supporting the display of 10x Visium and MERFISH data. It does not include statistical data visualization, only providing spatial plots of gene expression. Furthermore, Samui allows for the upload and overlay of multiple gene fluorescence images, primarily for intuitively displaying the spatial expression of genes but without quantitative analysis.

SpatialView is similar to Cellxgene VIP but is not as powerful or comprehensive. SpatialView is an R package that generates interfaces with interactive spatial plots and offers boxplot and heatmap data visualization methods. However, its interactive spatial plots do not support local comparison, and multiple spatial plots cannot be displayed simultaneously on one interface. While SpatialView supports cross-sample comparison, this is mainly reflected in boxplots and heatmaps, with limited flexibility. It supports reading processed h5ad, SPE, and Seurat objects.

SPATA2, STutility, and i-stLearn provide comprehensive R-based toolkits with advanced analysis features. SPATA2 is an excellent R package for downstream SRT analysis, capable of conducting SRT copy number variation (CNV) and pseudotime analysis. Although these functions are not implemented interactively, they can be displayed using Shiny applications. SPATA2 includes functions to generate visual interfaces for these results, though the generated interfaces are not very user-friendly, with somewhat rigid operations and limited functionality. SPATA2 typically converts Seurat or SPE objects into SPATA2 objects or uses SpaceRanger results for downstream analysis.

STutility is similar to SPATA2 and is also a downstream SRT analysis tool. It has SRT data processing capabilities and can directly process h5ad files or expression matrices output by SpaceRanger, similar to Seurat tools. Additionally, it can integrate multiple slices of the same sample to display 3D spatial plots.
i-stLearn is a tool that includes data processing and visualization. It can read raw data for processing and downstream analysis, conducted through code lines, and generate interactive interfaces for data sharing and analysis. Its data visualization interfaces mainly include clustered spatial plots, gene expression spatial plots, and ligand-receptor paired spatial plots. i-stLearn also offers an online analysis webpage that supports processed h5ad and SpaceRanger output files for online analysis. Overall, i-stLearn's interactive capabilities are not very robust or comprehensive, and its interactive interface lacks data visualization methods.
Loupe Browser is specifically designed for 10x Genomics data. Nowadays, Loupe Browser can annotate SRT spots, view gene spatial expression, and display spatial plots of gene expression. Users can interactively select spots in the spatial plot for statistical analysis, draw violin plots, and perform differential analysis. Despite its improved flexibility, the analysis operations remain limited, and it only supports loupe files.
TissUUmaps3 supports detailed visualization with a focus on multi-scale data integration. It can read h5ad files or CSV files containing gene expression and spatial information. Although its visualization primarily consists of spatial plots and lacks statistical data visualization methods, TissUUmaps3 supports the display of multi-sample spatial plots. Overall, it is an excellent spatial visualization tool.

SMDB and spatialLIBD focus more on specific datasets and provide centralized database functions. Besides offering databases, SMDB provides online SRT data visualization tools, mainly displaying spatial plots and boxplots. Its most notable feature is the ability to construct 3D brain structures using multiple brain SRT slices. Using SMDB requires providing TSV files of expression matrices and spatial location information.

ShinySRT is a lightweight SRT visualization tool that combines the strengths of existing SRT visualization tools while addressing their shortcomings. ShinySRT features a user-friendly interface with simple and intuitive operation. Installation is straightforward, and generating visualizations requires minimal coding, significantly lowering the barrier to entry. ShinySRT supports various mainstream SRT data types and can read and process raw data to produce visualizations. This process involves just one additional command beyond generating the interface. Researchers with limited bioinformatics expertise can use raw data outputs from SpaceRanger, run a simple Seurat workflow, and obtain complete SRT visualization interfaces.
ShinySRT offers a wide range of visualization methods, and users can customize the generated Shiny app code according to their needs. The key advantage of ShinySRT over other tools lies in its ability to display multiple samples and perform cross-sample comparisons. This feature is not limited to merely displaying multiple samples on the same interface; it also includes local comparisons across samples. ShinySRT excels in providing comprehensive multi-sample visualizations and cross-sample comparisons, reflected in both spatial plots and statistical analyses, enhancing the flexibility of the analysis.
Furthermore, ShinySRT considers additional variables beyond the two primary ones during data analysis, significantly increasing the flexibility of the analysis. These innovations in multi-sample display and cross-sample comparison were inspired by the strengths of other SRT visualization tools. We provide a demonstration of these features using a set of public SRT data in the supplementary materials (S1 File)(Quintana et al., 2022).
Overall, while ShinySRT does not introduce new technologies, it thoughtfully integrates and enhances existing capabilities to offer a more user-friendly and flexible tool for SRT data visualization and analysis.
 
### Reference
Bergenstråhle, J., Larsson, L., Lundeberg, J., 2020. Seamless integration of image and molecular analysis for spatial transcriptomics workflows. BMC Genomics 21(1), 482. https://doi.org/10.1186/s12864-020-06832-3.

Cao, R., Ling, Y., Meng, J., Jiang, A., Luo, R., He, Q., Li, A., Chen, Y., Zhang, Z., Liu, F., Li, Y., Zhang, G., 2023. SMDB: a Spatial Multimodal Data Browser. Nucleic Acids Research 51(W1), W553-W559. https://doi.org/10.1093/nar/gkad413.

Chaichontat, S., Annie, N., Nicholas, J.E., Leonardo, C.-T., Keri, M., Kristen, R.M., Stephanie, C.H., 2023. Performant web-based interactive visualization tool for spatially-resolved transcriptomics experiments. bioRxiv, 2023.2001.2028.525943. https://doi.org/10.1101/2023.01.28.525943.

Chitrasen, M., Aman, P., Lingxin, C., Lisa, M.A., Bridget, E.S., Beth, D., Christina, K., 2023. SpatialView: An interactive web application for visualization of multiple samples in spatial transcriptomics experiments. bioRxiv, 2023.2006.2013.544836. https://doi.org/10.1101/2023.06.13.544836.

Duy, P., Brad, B., Quan, H.N., 2023. i-stLearn: An interactive platform for spatial transcriptomics analysis. bioRxiv, 2023.2003.2027.534291. https://doi.org/10.1101/2023.03.27.534291.

Jan, K., Jasmin von, E., Vidhya, M.R., Paulina, W., Kevin, J., Juergen, B., Ulrich, G.H., Daniel, D., Oliver, S., Dieter Henrik, H., 2020. Inferring spatially transient gene expression pattern from spatial transcriptomic studies. bioRxiv, 2020.2010.2020.346544. https://doi.org/10.1101/2020.10.20.346544.

Kejie, L., Zhengyu, O., Yirui, C., Jacob, G., Dongdong, L., Michael, M., Will, C., David, S., Baohong, Z., 2022. Cellxgene VIP unleashes full power of interactive visualization and integrative analysis of scRNA-seq, spatial transcriptomics, and multiome data. bioRxiv, 2020.2008.2028.270652. https://doi.org/10.1101/2020.08.28.270652.

Pardo, B., Spangler, A., Weber, L.M., Page, S.C., Hicks, S.C., Jaffe, A.E., Martinowich, K., Maynard, K.R., Collado-Torres, L., 2022. spatialLIBD: an R/Bioconductor package to visualize spatially-resolved transcriptomics data. BMC Genomics 23(1), 434. https://doi.org/10.1186/s12864-022-08601-w.

Pielawski, N., Andersson, A., Avenel, C., Behanova, A., Chelebian, E., Klemm, A., Nysjö, F., Solorzano, L., Wählby, C., 2023. TissUUmaps 3: Improvements in interactive visualization, exploration, and quality assessment of large-scale spatial omics data. Heliyon 9(5). https://doi.org/10.1016/j.heliyon.2023.e15306.

Quintana, J.F., Chandrasegaran, P., Sinton, M.C., Briggs, E.M., Otto, T.D., Heslop, R., Bentley-Abbot, C., Loney, C., de Lecea, L., Mabbott, N.A., MacLeod, A., 2022. Single cell and spatial transcriptomic analyses reveal microglia-plasma cell crosstalk in the brain during Trypanosoma brucei infection. Nature Communications 13(1), 5752. https://doi.org/10.1038/s41467-022-33542-z.

